recipes/repo.rb

```diff
@@ -172,38 +177,20 @@
   mode '0600'
 end
 
-file "/home/#{agent_username}/.ssh/gpg_private_key.sec" do
-  owner agent_username
-  group agent_username
-  mode '0600'
-  content gpg_key['private_key']
-end
-file "/home/#{agent_username}/.ssh/gpg_public_key.pub" do
-  owner agent_username
-  group agent_username
-  mode '0644'
-  content gpg_key['public_key']
-end
```

Comment:
We were just discussing deployment strategy offline but do you think it's worth changing these to
```ruby
file "/home/#{agent_username}/.ssh/gpg_private_key.sec" do
  action :delete
end
```

so that manual steps are not required? If we wanted to do it as a two step process we could leave these file resources un-touched for the initial deployment and then remove them in a follow-up change but I would be worried that if we just delete these resources we'll forget to clean up the old files via chef.

Reply:
We could do this, but it still wouldn't be enough. We need to remove the private key from the `jenkins-agent` keyring as well. Doing that directly conflicts with the switch to use the `gpg-vault`, so special consideration would be necessary to ensure that the `gpg-vault` switch happened after the "upgrade" logic to drop the keypair files and remove the private key from the keyring.

I can pursue adding this "upgrade" logic if you like.